src/Trinity.Core/Utilities/MISC/Misc.cs

```diff
@@ -13,37 +13,28 @@ internal class Misc<T>
     {
         public static string ToString(ICollection<T> collection, int indent = 0, string separator = " ")
         {
-            string indent_space = "";
-            for (int i = 0; i < indent; i++)
-            {
-                indent_space += " ";
-            }
-            string ret = "";
-            foreach (T t in collection)
-            {
-                ret += indent_space + t.ToString() + separator;
-            }
-            if (ret.LastIndexOf(separator, StringComparison.Ordinal) != -1) //make sure the input is not an empty collection
-            {
-                ret = ret.Substring(0, ret.LastIndexOf(separator, StringComparison.Ordinal)); //remove the last separator
-            }
-            return ret;
+            string indent_space = new string(' ', indent);
+            return string.Join(separator, collection.Select(c => indent_space + c));
         }
     }
 
     internal class Misc
     {
         public static void DisplayByteArray(byte[] bytes)
         {
-            for (int i = 0; i < bytes.Length; )
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < bytes.Length; i++)
             {
-                for (int j = 0; j < 8; j++)
+                sb.Append(bytes[i++]);

```

Comment:
with iterator incremented in the for loop it should not be pushed forward again here ;)